### PR TITLE
feat: support `chrome.scripting` extension  APIs

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

Closes https://github.com/electron/electron/issues/37876.

This PR adds support for [`chrome.scripting`](https://developer.chrome.com/docs/extensions/reference/scripting).

From reference docs: 

> `chrome.scripting` allows API to inject JavaScript and CSS into websites. This is similar to what you can do with [content scripts](https://developer.chrome.com/docs/extensions/mv3/content_scripts/). But by using the [chrome.scripting](https://developer.chrome.com/docs/extensions/reference/scripting/) namespace, extensions can make decisions at runtime. 

This represents an important step towards proper support of Manifest V3. Specifically -  `chrome.scripting` supersedes `chrome.tabs.executeScript`, which is gone in Manifest V3. It's commonly used in several DevTools extensions, including [React DevTools](https://github.com/facebook/react/tree/main/packages/react-devtools).

The source code added here is essentially unchanged from [upstream](https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/extensions/api/scripting/scripting_api.h) - we need to add it to our api definitions because it lives upstream in `chrome/browser/extensions/api`, which we do not depend on (and shouldn't, since many of the api implementations therein involve concepts not transferable to Electron without significant modification). 

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [x] relevant documentation, tutorials, templates and examples are changed or added
- [x] [PR release notes](https://github.com/electron/clerk/blob/main/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/main/README.md#examples).

#### Release Notes

Notes: Added support for `chrome.scripting` extension APIs.

### CODE REVIEW

File: shell/browser/extensions/api/scripting/scripting_api.h
Comment (codebytere) [inline shell/browser/extensions/api/scripting/scripting_api.h:1]: Ideally we should remove this but our extensions paradigm requires this header being in the `electron::api::extensions` namespace.

**Review by zcbenz — APPROVED** [review]
API LGTM

**Review by jkleinsc — APPROVED** [review]
API LGTM

**Comment by codebytere** [comment]
@jasonfill i've tried to tune up [this doc](https://github.com/electron/electron/blob/f0ad357af207a7f8b4a425bbad0e23d15c92fbbc/docs/api/extensions.md#L1) to reflect current support - what would help that doc be more effective for you?

**Pull request merged**